File: src/app/models/item.rb
# frozen_string_literal: true

class Item
  attr_accessor :id, :code, :name, :price

  def self.header
    %w[id code name price]
  end

  def initialize(properties = {})
    @id = properties[:id]
    @name = properties[:name]
    @code = properties[:code]
    @price = properties[:price]
  end

  def to_row
    [id, code, name, price]
  end
end

File: src/items/Item.php
<?php

class Item {
    protected $itemCode;
    protected $itemName;
    protected $itemPrice;
    protected $itemOneFree;

    public function __construct($code, $name, $price, $oneFree = null)
    {
        $this->itemCode = $code;
        $this->itemName = $name;
        $this->itemPrice = $price;
        $this->itemOneFree = $oneFree;
    }

    public function getItemCode() {
        return $this->itemCode;
    }

    public function getItemName() {
        return $this->itemName;
    }

    public function getItemPrice() {
        return $this->itemPrice;
    }

    public function setItemPrice($price){
        $this->itemPrice = $price;
    }

    public function getItemOnefree(){
        return $this->itemOneFree;
    }

}